File: src/Singleton/singleton.py
class Singleton:
    _instance = None

    def __new__(cls, *args, **kwargs):
        if not cls._instance:
            cls._instance = super(Singleton, cls).__new__(cls, *args, **kwargs)
        return cls._instance

# Usage
singleton1 = Singleton()
singleton2 = Singleton()

print(singleton1 is singleton2)  # Output: True

File: src/Singleton/singleton.pl
package Singleton;

my $instance;

sub new {
    my $class = shift;
    unless (defined $instance) {
        $instance = bless {}, $class;
    }
    return $instance;
}

# Usage
my $singleton1 = Singleton->new();
my $singleton2 = Singleton->new();

print $singleton1 == $singleton2 ? "true\n" : "false\n";  # Output: true
